### ISSUE TITLE
gotta start: Fallback to any open port when 3000 is currently in use

### ISSUE THREAD

**Issue by tylerklose** [gotta-dev/gotta#31]
If you're trying to get going locally with `gotta start` and there's something already running on `3000`, we shouldn't just error out and hang. Instead, we should search for an open port and then prompt the user to press `<enter>` or something to spin up your `gotta` app on that port

**Comment by parruda** [comment]
This could cause some nasty behaviour in production.
But I agree the current error handling is poor. `gotta start` should send a SIGTERM to all child processes if puma errors because the port is already in use.
Maybe we could rescue `Errno::EADDRINUSE` here: https://github.com/gotta-dev/gotta/blob/master/lib/gotta/serve/cli/serve.rb#L145

**Comment by tylerklose** [comment]
I don't think this be used in production. Locally only. This is totally taking a page out of `yarn`/`React`'s playbook. I think it's a nice DX: https://take.ms/OSiQK

**Comment by parruda** [comment]
Yeah I agree it would be nice to have it behave like that only through `gotta start`. But I think `gotta serve` should just fail and quit when the port is already in use.

That way `gotta start` could just restart `gotta serve` on another port.
